frontend/src/app/hooks/useAuth.ts

```diff
@@ -0,0 +1,87 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { useRouter } from "next/navigation";
+
+export interface AuthUser {
+  userId: string;
+  firstName: string;
+  lastName: string;
+  email: string;
+  role: string;
+}
+
+export function useAuth() {
```

Comment:
The useAuth function is specified in /src/contexts/auth-context.tsx where it uses AuthContext that implements all the auth state variables, it has been implemented twice and is causing some issues with logging in and out & state re-renders. Please combine into one useAuth hook.

Reply:
Has be deleted, artefact of merge